### DISCUSSION TITLE
How to enable zookeeper authentication?

### DISCUSSION THREAD

**Discussion by HaoYang670** [apache/kyuubi#5123] [Q&A]
Is there a way to enable zookeeper authentication, so that we can preserve the engine info?

**Comment by pan3793** [comment]
> Is there a way to enable zookeeper authentication

In short, `kyuubi.ha.zookeeper.auth.type=KERBEROS`

And you may want to set a dedicated principal and keytab for zookeeper, `kyuubi.ha.zookeeper.auth.principal` `kyuubi.ha.zookeeper.auth.keytab`

**Reply by HaoYang670** [reply]
This doesn't work for kyuubi's emdedded zookeeeper, right?

**Reply by pan3793** [reply]
Yea, the embedded zookeeper is only for demonstration purposes, it's not supposed for users to use it for production. We only expose a subset of its configuration like binding port, data dir, and Kerberos-related configurations are not included.
